### PR TITLE
Book library basic functionality

### PR DESCRIPTION
**Pull request by AbdelrhmanAmin**
I've added those features:
 - New book creation.
 - Storage of the boos.
 - Set books as read or unread.
 - Delete books

### CODE REVIEW

**Comment by sumancrest0001** [comment]
**Status: Changes required :gear:**

**Hello @AbdelrhmanAmin,**
Thank you for submitting this project. You have implemented all the coding requirements from the specifications. But there are few things that need to be addressed before moving ahead. Let's have a look at them.

**[Required]** guideline is not considered while creating the PR. It should contain a title and a short description of the features implemented.  [Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_prs.png)

**[Required]** Commit messages should be capitalized and descriptive. 

**[Required]** You have added the workflow but you seem to forget to add linter configurations. Please check the [reference](https://github.com/microverseinc/linters-config/tree/master/javascript)

File: README.md [hunk 1]
```diff
@@ -1 +1,56 @@
...
+1- Clone the repository
+```
+    git clone https://github.com/AbdelrhmanAmin/Library-js.git
+```
+
+2- Open the folder on VS Code. 
+```
+    cd Book-js
+    code .
+```
```
Comment (sumancrest0001) [inline README.md]: **[Required]** It should be **Library-js**.

**Review by sumancrest0001 — CHANGES_REQUESTED** [review]
**Please make the required changes and submit for another code review.** 

**If you have any queries regarding this review, message me on slack.
Suman Shrestha**

**Happy coding !!!**

**Comment by AbdelrhmanAmin** [comment]
@sumancrest0001  I've already added the linters.

File: index.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
+let myLibrary = [];
+const auth = document.getElementById('author');
+const title = document.getElementById('title');
+const pages = document.getElementById('pages');
+const read = document.getElementById('read');
+const btn = document.getElementById('submit');
+const table = document.getElementById('table');
+
+function getBooks() {
```
Comment (christianotieno) [inline index.js]: ## Implement function with ES6 arrow function
- [ ] Implement this `getBooks` function with ES6 arrow function
```suggestion
const getBooks = () => {
```

After the book has been created, there is no way to display the books in the `localStorage` as there is no way javascript can iterate over the array properly. This is the error we currently have. 👇 
library-book-creation-error
- [ ] Kindly fix this issue, thank you!

File: index.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  let data = localStorage.getItem('library');
+  data = JSON.parse(data);
+  return data;
+}
+function setBooks() {
```
Comment (christianotieno) [inline index.js:14]: ## Implement function with ES6 arrow function
- [ ] Implement this `setBooks` function with ES6 arrow function
```suggestion
const setBooks = () => {
```

File: index.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+function Book(author, title, pages, read = false) {
+  this.author = author;
+  this.title = title;
+  this.pages = pages;
+  this.read = read;
+}
+
+function addBookToLibrary(arr, book) {
```
Comment (christianotieno) [inline index.js:26]: ## Implement function with ES6 arrow function
- [ ] Implement this `addBookToLibrary` function with ES6 arrow function
```suggestion
const addBookToLibrary = (arr, book) => {
```

File: index.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+btn.addEventListener('click', (e) => {
+  e.preventDefault();
+  if (auth.value !== '' && title.value !== '' && pages.value !== '') {
+    const book = new Book(auth.value, title.value, pages.value, read.checked);
+    addBookToLibrary(myLibrary, book);
+  }
+});
+
+function displayEachBook() {
```
Comment (christianotieno) [inline index.js:50]: ## Implement function with ES6 arrow function
- [ ] Implement this `displayEachBook` function with ES6 arrow function
```suggestion
const displayEachBook = () => {
```

File: index.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    tdBtn2.appendChild(readBtn);
+    tdAuthor.innerHTML = book.author;
+    tdTitle.innerHTML = book.title;
+    tdPages.innerHTML = book.pages;
+    tr.appendChild(tdAuthor);
+    tr.appendChild(tdTitle);
+    tr.appendChild(tdPages);
+    tr.appendChild(tdBtn2);
+    tr.appendChild(tdBtn);
+    table.appendChild(tr);
```
Comment (christianotieno) [inline index.js]: ## Modularise this module. 

- [ ] Kindly keep application logic separated from DOM manipulation tasks ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/dom_logic_separation_requirement.png) from Pathwright)
- [ ] While at it, try and keep the number of changes/updates to the DOM as low as possible, they are costly for the browser ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/dom_updates_requirement.png) from Pathwright)

**Review by christianotieno — CHANGES_REQUESTED** [review]
## Changes required ♻️ 
Hello @AbdelrhmanAmin 👋 

The progress made in this project is commendable. We, however, do have some issues to work on before proceeding. 👇 

The changes touch on the use of ES6 arrow functions, as it was already introduced to you in the [previous lesson](https://microverse.pathwright.com/library/javascript-module-v10-copy/166786/path/step/97635162/)([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/es6_requirement.png)), and an issue about an error with book creation.

### Happy coding 💻 
#### Christian Otieno (CR)

**Review by peterrobert — APPROVED** [review]
### APPROVED 👍🏿 💯 
Hello, @AbdelrhmanAmin  very nice project. I love the design. I also love the way you have easily kept your code simple and working keep up 👍🏿 

### on to the next project 👍🏿  
### HAPPY CODING 💻

**Pull request merged**